src/go/rpk/pkg/cli/cmd/acl/user.go

```diff
@@ -97,11 +98,38 @@ acl help text for more info.
 			} else {
 				out.Die("missing required username argument")
 			}
-			if pass == "" {
-				if passOld == "" { // backcompat
-					out.Die("missing required --password")
+
+			// Redpanda added support for using our Kafka SASL
+			// credentials for basic auth. We use --password on
+			// commands to set the Kafka SASL password, but we also
+			// use --password here to specify the new user
+			// password. Historically, this was fine.
+			//
+			// Now, we support --new-password AND --password
+			// (--new-password was the original flag), and we add
+			// the short form -p to --new-password. Previously, -p
+			// was on --password, which meant -p and --password set
+			// the same value and we could not tell.
+			//
+			// Now, if we detect --user, we require --new-password
+			// (or -p). If we only see --password (i.e. only "pass"
+			// is set), we fail.
+			//
+			// See #6360.
+			//
+			// Better long term is to switch all configuration
+			// setting flags to -X.
+			for _, arg := range os.Args {
+				const userFlag = "--" + config.FlagSASLUser
+				if arg == userFlag {
+					if newPass == "" {
+						out.Die("unable to create user with when using basic auth, use --new-password to specify the new user's password")
+					}
```

Comment:
Instead of iterating over the `os.Args` wdyt of using cobra's `cmd.Flag()` method?

```go
userFlag := cmd.Flag(config.FlagSASLUser).Value.String()
if userFlag == "" && newPass == "" {
	out.Die("unable to create user with when using basic auth, use --new-password to specify the new user's password")
}
```

Reply:
This is a great idea!